libbeat/licenser/elastic_fetcher.go

```diff
@@ -0,0 +1,140 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package licenser
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+const xPackURL = "/_xpack"
+
+// params defaults query parameters to send to the '_xpack' endpoint by default we only need
+// machine parseable data.
+var params = map[string]string{
+	"human": "false",
+}
+
+var stateLookup = map[string]State{
+	"inactive": Inactive,
+	"active":   Active,
+}
+
+var licenseLookup = map[string]LicenseType{
+	"oss":      OSS,
+	"basic":    Basic,
+	"gold":     Gold,
+	"platinum": Platinum,
+}
+
+// UnmarshalJSON takes a bytes array and convert it to the appropriate license type.
+func (t *LicenseType) UnmarshalJSON(b []byte) error {
+	if len(b) <= 2 {
+		return fmt.Errorf("invalid string for license type, received: '%s'", string(b))
+	}
+	s := string(b[1 : len(b)-1])
+	license, ok := licenseLookup[s]
+	if ok {
```

Comment:
Nit: Normally we check for the error an return the error inside the if clause and the flow to the end of the method is the non error flow.